src/Config.hs

```diff
@@ -101,40 +271,89 @@ instance Read LogLevel where
   readsPrec _ _ = []
 
 -- | Parse program options from the command line and the process environment.
-parseOptionsFromEnvAndCli :: [EnvVar] -> IO Options
-parseOptionsFromEnvAndCli envVars =
-  let envFlags = parseEnvFlags envVars
-      parser = optionsParserWithInfo envFlags envVars
-  in OptParse.execParser parser
+parseOptionsFromEnvAndCli :: [EnvVar] -> [[EnvVar]] -> IO Options
+parseOptionsFromEnvAndCli localEnvVars envFileSettings =
+  let eLocalEnvFlagsOptions = validateCopyAddr $ parseEnvFlags localEnvVars
+      eEnvFileSettingsOptions = map (validateCopyAddr . parseEnvFlags) envFileSettings
+  in do
+    eParseResult <- validateCopyAddr <$> OptParse.execParser optionsParserWithInfo
+    let results = eEnvFileSettingsOptions ++ [eLocalEnvFlagsOptions, eParseResult]
+    if any isLeft results then
+      die ("[ERROR] " ++ unlines (map show $ lefts results))
+    else return $ foldl (flip mergeOptions) defaultOptions (rights results)
 
 -- | Parses behavior flags from a list of environment variables. If an
 -- environment variable corresponding to the flag is set to @"true"@ or
 -- @"false"@, we use that as the default on the corresponding CLI option.
 --
 -- If these variables aren't present, we default to @False@. We print an error
 -- if they're set to anything else than @"true"@ or @"false"@.
-parseEnvFlags :: [EnvVar] -> EnvFlags
+parseEnvFlags :: [EnvVar] -> Options
 parseEnvFlags envVars
-  = EnvFlags
-  { efConnectTls = lookupEnvFlag "VAULTENV_CONNECT_TLS"
-  , efValidateCerts = lookupEnvFlag "VAULTENV_VALIDATE_CERTS"
-  , efInheritEnv = lookupEnvFlag "VAULTENV_INHERIT_ENV"
-  , efUsePath = lookupEnvFlag "VAULTENV_USE_PATH"
+  = emptyOptions
+  { oVaultHost      = lookupEnvString   "VAULT_HOST"
+  , oVaultPort      = lookupEnvInt      "VAULT_PORT"
+  , oVaultAddr      = lookupEnvString   "VAULT_ADDR"  
+  , oVaultToken     = lookupEnvString   "VAULT_TOKEN"
+  , oSecretFile     = lookupEnvString   "VAULTENV_SECRETS_FILE"
+  , oCmd            = lookupEnvString   "CMD"
+  , oArgs           = lookupStringList  "ARGS..."
+  , oConnectTls     = lookupEnvFlag     "VAULTENV_CONNECT_TLS"
+  , oValidateCerts  = lookupEnvFlag     "VAULTENV_VALIDATE_CERTS"
+  , oInheritEnv     = lookupEnvFlag     "VAULTENV_INHERIT_ENV"
+  , oRetryBaseDelay = MilliSeconds <$> lookupEnvInt      "VAULTENV_RETRY_BASE_DELAY"
+  , oRetryAttempts  = lookupEnvInt      "VAULTENV_RETRY_ATTEMPTS"
+  , oLogLevel       = lookupEnvLogLevel "VAULTENV_LOG_LEVEL"
+  , oUsePath        = lookupEnvFlag     "VAULTENV_USE_PATH"
   }
   where
+    err :: String -> a
+    err key = errorWithoutStackTrace $ "[ERROR]: Invalid value for environment variable " ++ key
+    lookupEnvString key = lookup key envVars
```

Comment:
This will not work, as these functions use the local variable envVars, which needs to be passed explicitly in such a case. But I added the type signatures and comments.